### PR TITLE
Task yc/j14 add header

### PR DESCRIPTION
**Pull request by EugeniusGR**
Header added
image
image

### CODE REVIEW

**Comment by EugeniusGR** [comment]
[Header Component](https://trello.com/c/QGleHfrZ/14-header-component)

File: frontend/public/index.html [hunk 1]
```diff
@@ -25,6 +25,12 @@
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
     <title>React Redux App</title>
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
```
Comment (ghost) [inline frontend/public/index.html]:
```suggestion
```

File: frontend/src/components/app/app.tsx [hunk 2]
```diff
@@ -6,12 +6,15 @@ import SignUp from 'components/sign-up/sign-up';
 import { Link, Route, Switch } from 'components/common/common';
 import logo from 'assets/img/logo.svg';
 
+import Header from 'components/header/header';
```
Comment (ghost) [inline frontend/src/components/app/app.tsx]: move to common components

File: frontend/src/components/header/header.css [hunk 3]
```diff
@@ -0,0 +1,129 @@
+* {
+  margin: 0;
+}
```
Comment (ghost) [inline frontend/src/components/header/header.css]:
```suggestion
```

File: frontend/src/components/header/header.css [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  align-items: center;
+  width: 1900px;
+  height: 104px;
+  background-color: var(--color-header-background);
+  color: #fff;
+  text-align: left;
+  font-family: 'Raleway';
+}
+
+.LogoContainer {
```
Comment (ghost) [inline frontend/src/components/header/header.css]: why it it capitalize ? remove

File: frontend/src/components/header/header.css [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+.container {
+  display: flex;
+  align-items: center;
+  width: 1900px;
+  height: 104px;
+  background-color: var(--color-header-background);
+  color: #fff;
+  text-align: left;
+  font-family: 'Raleway';
```
Comment (ghost) [inline frontend/src/components/header/header.css]: add it to global styles

File: frontend/src/components/header/header.css [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+  margin-right: 4px;
+  width: 22px;
+  height: 22px;
+  background-color: #aeaeae;
+  border-radius: 50%;
+}
+
+.navigation {
+  display: flex;
+  width: 325px;
```
Comment (ghost) [inline frontend/src/components/header/header.css]:
```suggestion
```

File: frontend/src/components/header/header.css [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+  margin-left: 764px;
+  width: 173px;
+  height: 43px;
+  background-color: var(--color-button-background);
+  border: 1px solid var(--color-button-background);
+  border-radius: 7px;
+
+  cursor: pointer;
+}
```
Comment (ghost) [inline frontend/src/components/header/header.css]:
```suggestion
```
don't style the button

File: frontend/src/components/header/header.tsx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      <div className="LogoContainer">
+        <div className="LogoContainerLogo"></div>
+        <span>Logo</span>
+      </div>
+
+      <div className="navigation">
+        <span className="spanNavigation activeSpanNavigation">Podcasts</span>
+        <span className="spanNavigation">Streaming</span>
+        <span className="spanNavigation">Setting</span>
+      </div>
```
Comment (ghost) [inline frontend/src/components/header/header.tsx]: use `ul` with `li`
also user Link component from common

File: frontend/src/store/auth/reducer.ts [hunk 9]
```diff
@@ -1,13 +1,15 @@
 import { createReducer } from '@reduxjs/toolkit';
 import { DataStatus } from 'common/enums/enums';
-import { signUp } from './actions';
+import { signUp, User } from './actions';
 
 type State = {
   dataStatus: DataStatus;
+  sessionUser: User | null;
```
Comment (ghost) [inline frontend/src/store/auth/reducer.ts]:
```suggestion
  user: User | null;
```

File: frontend/src/store/auth/reducer.ts [hunk 10]
```diff
@@ -1,13 +1,15 @@
...
 };
 
 const initialState: State = {
   dataStatus: DataStatus.IDLE,
+  sessionUser: null,
```
Comment (ghost) [inline frontend/src/store/auth/reducer.ts]:
```suggestion
  user: null,
```

File: frontend/src/store/auth/actions.ts [hunk 11]
```diff
@@ -11,3 +11,4 @@ const signUp = createAsyncThunk<Promise<User>, UserCreatePayload, AsyncThunkConf
 });
 
 export { signUp };
+export type { User };
```
Comment (ghost) [inline frontend/src/store/auth/actions.ts]:
```suggestion
```

File: frontend/src/store/auth/reducer.ts [hunk 12]
```diff
@@ -1,13 +1,15 @@
 import { createReducer } from '@reduxjs/toolkit';
 import { DataStatus } from 'common/enums/enums';
-import { signUp } from './actions';
+import { signUp, User } from './actions';
```
Comment (ghost) [inline frontend/src/store/auth/reducer.ts]:
```suggestion
import { signUp } from './actions';
```
why? `common/types/types` ?

File: frontend/src/components/header/header.tsx [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+            <div className="notificationActive"></div>
+          </div>
+
+          <div className="profile">
+            <img className="profileAvatar" src={avatar} alt="Kiwi standing on oval" />
+          </div>
+        </div>
+      ) : (
+        <div className="signIn">
+          <a href="#" className="signInText">
```
Comment (ghost) [inline frontend/src/components/header/header.tsx]: use Link from common components

File: frontend/src/components/header/header.tsx [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+              <path
+                fill="currentColor"
+                d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
+              ></path>
+            </svg>
+            <div className="notificationActive"></div>
+          </div>
+
+          <div className="profile">
+            <img className="profileAvatar" src={avatar} alt="Kiwi standing on oval" />
```
Comment (ghost) [inline frontend/src/components/header/header.tsx]: add width/height/loading attribute

File: frontend/src/components/header/header.tsx [hunk 15]
```diff
@@ -0,0 +1,66 @@
+import './header.css';
+import { useSelector } from 'hooks/hooks';
+import { RootState } from '../../common/types/types';
+import avatar from '../../assets/img/default-user-avatar.svg';
```
Comment (ghost) [inline frontend/src/components/header/header.tsx]:
```suggestion
import defaultAvatar from 'assets/img/default-user-avatar.svg';
```

File: frontend/src/components/header/header.tsx [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+        <span>Logo</span>
+      </div>
+
+      <div className="navigation">
+        <span className="spanNavigation activeSpanNavigation">Podcasts</span>
+        <span className="spanNavigation">Streaming</span>
+        <span className="spanNavigation">Setting</span>
+      </div>
+
+      {sessionUser ? (
```
Comment (ghost) [inline frontend/src/components/header/header.tsx]:
```suggestion
     const hasUser = Boolean(sessionUser);

      {hasUser ? (
```

File: frontend/src/components/header/header.tsx [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+
+const Header: React.FC = () => {
+  const { sessionUser } = useSelector(({ auth }: RootState) => ({
+    sessionUser: auth.sessionUser,
+  }));
+
+  return (
+    <div className="container">
+      <div className="LogoContainer">
+        <div className="LogoContainerLogo"></div>
```
Comment (ghost) [inline frontend/src/components/header/header.tsx]:
```suggestion
      <div className="logoContainer">
        <div className="logoContainerLogo"></div>
```

File: frontend/src/components/header/header.tsx [hunk 18]
```diff
@@ -0,0 +1,66 @@
+import './header.css';
```
Comment (ghost) [inline frontend/src/components/header/header.tsx]: move it to bottom of imports

use scss-modules

File: frontend/src/components/header/header.tsx [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+
+      <div className="navigation">
+        <span className="spanNavigation activeSpanNavigation">Podcasts</span>
+        <span className="spanNavigation">Streaming</span>
+        <span className="spanNavigation">Setting</span>
+      </div>
+
+      {sessionUser ? (
+        <div className="userInfo">
+          <button className="userInfoCreateButton">
```
Comment (ghost) [inline frontend/src/components/header/header.tsx]: use Button from common components

File: frontend/src/components/header/header.tsx [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+              xmlns="http://www.w3.org/2000/svg"
+              viewBox="0 0 448 512"
+            >
+              <path
+                fill="currentColor"
+                d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
+              ></path>
+            </svg>
+            <div className="notificationActive"></div>
+          </div>
```
Comment (ghost) [inline frontend/src/components/header/header.tsx]:
```suggestion
```
don't forget remove styles

File: frontend/src/components/header/header.tsx [hunk 21]
```diff
@@ -0,0 +1,66 @@
...
+import { RootState } from '../../common/types/types';
+import avatar from '../../assets/img/default-user-avatar.svg';
+
+const Header: React.FC = () => {
+  const { sessionUser } = useSelector(({ auth }: RootState) => ({
+    sessionUser: auth.sessionUser,
+  }));
+
+  return (
+    <div className="container">
```
Comment (ghost) [inline frontend/src/components/header/header.tsx]:
```suggestion
    <header className="container">
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/assets/css/styles.scss [hunk 22]
```diff
@@ -1 +1,5 @@
 @import './variables';
+
+body{
```
Comment (ghost) [inline frontend/src/assets/css/styles.scss]:
```suggestion
body {
```

File: frontend/src/components/common/header/header.module.css [hunk 23]
```diff
@@ -0,0 +1,87 @@
...
+
+.profile {
+  width: 40px;
+  height: 40px;
+  margin-left: 84px;
+  cursor: pointer;
+}
+.profileAvatar {
+  max-width: 100%;
+  max-height: 100%;
```
Comment (ghost) [inline frontend/src/components/common/header/header.module.css]: img {
    max-width: 100%;
    height: auto;
}
please, add it to global style

```suggestion
```

File: frontend/src/components/common/header/header.tsx [hunk 24]
```diff
@@ -0,0 +1,46 @@
...
+
+  return (
+    <header className={styles.container}>
+      <div className={styles.logoContainer}>
+        <div className={styles.logoContainerLogo}></div>
+        <span>Logo</span>
+      </div>
+
+      <ul className={styles.navigation}>
+        <li className={` ${styles.liNavigation} ${styles.activeLiNavigation}`}>Podcasts</li>
```
Comment (ghost) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
        <li className={styles.liNavigation}>Podcasts</li>
```
Don't need it for now
Also don't forget to remove styles

File: frontend/src/components/common/header/header.tsx [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+        <li className={styles.liNavigation}>Streaming</li>
+        <li className={styles.liNavigation}>Setting</li>
+      </ul>
+
+      {!hasUser ? (
```
Comment (ghost) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
      {hasUser ? (
```
right? 🤔

File: frontend/src/store/auth/actions.ts [hunk 26]
```diff
@@ -2,7 +2,7 @@ import { createAsyncThunk } from '@reduxjs/toolkit';
 import { UserCreatePayload, AsyncThunkConfig, User } from 'common/types/types';
 import { ActionType } from './common';
 
-const signUp = createAsyncThunk<Promise<User>, UserCreatePayload, AsyncThunkConfig>
+const signUp = createAsyncThunk<Promise<User>,UserCreatePayload,AsyncThunkConfig>
```
Comment (ghost) [inline frontend/src/store/auth/actions.ts]: why?

File: frontend/src/components/common/header/header.tsx [hunk 27]
```diff
@@ -0,0 +1,46 @@
+import { useSelector } from 'hooks/hooks';
+import { AppRoute, ButtonType } from 'common/enums/enums';
+import { RootState } from '../../../common/types/types';
+import { Button, Link } from '../common';
+import defaultAvatar  from '../../../assets/img/default-user-avatar.svg';
```
Comment (ghost) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
import { RootState } from 'common/types/types';
import { Button, Link } from 'components/common';
import defaultAvatar  from 'assets/img/default-user-avatar.svg';
```
always use aliases

File: frontend/src/components/common/header/header.tsx [hunk 28]
```diff
@@ -0,0 +1,46 @@
...
+import styles from './header.module.css';
+
+const Header: React.FC = () => {
+  const { user } = useSelector(({ auth }: RootState) => ({
```
Comment (ghost) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
  const { user } = useAppSelector(({ auth }: RootState) => ({
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/header/header.tsx [hunk 29]
```diff
@@ -0,0 +1,46 @@
+import { useAppSelector } from 'hooks/hooks';
+import { AppRoute, ButtonType } from 'common/enums/enums';
+import { RootState } from 'common/types/types';
+import { Button, Link } from 'components/common/common';
+import defaultAvatar  from 'assets/img/default-user-avatar.svg';
+import styles from './header.module.css';
```
Comment (what1s1ove) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
import styles from './styles.module.css';
```

File: frontend/src/components/common/header/header.tsx (hunk 25 above)
Comment (what1s1ove) [reply frontend/src/components/common/header/header.tsx]: isn't fixed

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/header/header.tsx [hunk 30]
```diff
@@ -0,0 +1,46 @@
...
+
+      {!hasUser ? (
+        <div className={styles.userInfo}>
+          <Button label="+ Create Podcast" type={ButtonType.BUTTON}/>
+          <div className={styles.profile}>
+            <img className={styles.profileAvatar} src={defaultAvatar } width="40px" height="40px" alt="avatar" loading="lazy" />
+          </div>
+        </div>
+      ) : (
+        <div className="signIn">
```
Comment (what1s1ove) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
        <div>
```
or add styles-module

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/link/link.tsx [hunk 31]
```diff
@@ -3,10 +3,13 @@ import { AppRoute } from 'common/enums/enums';
 
 type Props = {
   to: AppRoute;
+  style: string;
```
Comment (what1s1ove) [inline frontend/src/components/common/link/link.tsx]:
```suggestion
  className?: string;
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]
👍

**Pull request merged**